### ISSUE TITLE
Don't modify tsconfig.json if there is no filesGlob property present

### ISSUE THREAD

**Issue by awaltert** [wjohnsto/tsconfig-glob#10]
**Situation**: the actual implementation adds the `files` property with an empty array to the `tsconfig.json`, although there is no `filesGlob` property present in the original `tsconfig.json`.

**Error**: In our project, we have an exclude property set which cannot be used in conjunction with the files property. Furthermore there is no `filesGlob` property defined in the original configuration file.  
Because of the behavior described above, the TypeScript compiler doesn’t find any ts files anymore.

> The "files" property cannot be used in conjunction with the "exclude" property. If both are specified then the"files" property takes precedence.

This behavior is described in detail [here](http://www.typescriptlang.org/docs/handbook/tsconfig.json.html#details). 

**Steps to repoduce**: In a sample project we use tsconfig-glob in a grunt-task to compile with the following `tsconfig.json`:  

``` json
{
  "compilerOptions": {
    "listFiles": false,
    "target": "es5",
    "module": "commonjs",
    "removeComments": false,
    "declaration": true,
    "sourceMap": true
  },
  "exclude": [
    "node_modules",
    "jspm_packages"
  ]
}
```

after running the grunt-task with tsconfig-glob we get the resulting `tsconfig.json`:

``` json
{
  "compilerOptions": {
    "listFiles": false,
    "target": "es5",
    "module": "commonjs",
    "removeComments": false,
    "declaration": true,
    "sourceMap": true
  },
  "exclude": [
    "node_modules",
    "jspm_packages"
  ],
  "files": []
}
```

The expected behavior would be:  
If the `filesGlob` property is absent, `tsconfig-glob` shouldn't append and set the files property.

**Comment by wjohnsto** [comment]
This does appear to be an issue. However if you're using "exlude" then I ask why you're also using `tsconfig-glob`? `tsconfig-glob` doesn't serve any purpose other than to look at `filesGlob` and fill out `files`.

**Comment by awaltert** [comment]
You are right that `tsconfig-glob` has the described purpose. Please consider that `tsconfig-glob` module won’t only be called directly, but also transitively from other modules. To be more concrete:  
I found this regression when looking at the module [grunt-typescript-using-tsconfig](https://github.com/gilamran/grunt-typescript-using-tsconfig). 
This module has `tsconfig-glob` defined as dependency. We use that module in our build chain to compile our projects to different targets. On every compile it delegates to `tsconfig-glob`. This call results in _defect_ `tsconfig.json` files, because `tsconfig-glob` always changes the corresponding config file. In our context, defect means, that the modified `tsconfig.json` files can’t be used anymore with plain `tsc`.  
In my opinion, the `tsconfig-glob` module has the responsibility to decide and modify the configuration files if necessary. Nevertheless it’s semantically incorrect to modify the `tsconfig.json` on every run if there is no `fileGlob` property present.  
Please consider the PR #11 for this issue. You will see that my modification has minimal implications to the runtime behavior for this module.

**Comment by wjohnsto** [comment]
Your fix is in v0.4.3
